### PR TITLE
Add `pipeline-groovy-lib`

### PR DESCRIPTION
**Pull request by jglick**
Supersedes #1166. See [JENKINS-57023](https://issues.jenkins.io/browse/JENKINS-57023), inspired by gratuitous dependencies in test classpaths: https://github.com/jenkinsci/bom/pull/1159#issuecomment-1134862481

Picks up:
* https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/172
* https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/530
* https://github.com/jenkinsci/workflow-multibranch-plugin/pull/180

Might also benefit from test-only dep fixes:
* https://github.com/jenkinsci/configuration-as-code-plugin/pull/1973
* https://github.com/jenkinsci/git-plugin/pull/1274
* https://github.com/jenkinsci/github-branch-source-plugin/pull/565

If all of those are integrated, I believe we could remove `workflow-cps-global-lib` from the managed set. At that point we may as well remove `git-server` as well, and could consider removing `sshd`.

### CODE REVIEW

File: bom-weekly/pom.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
         <checks-api.version>1.7.4</checks-api.version>
         <configuration-as-code-plugin.version>1429.v09b_044a_c93de</configuration-as-code-plugin.version>
         <git-plugin.version>4.11.3</git-plugin.version>
-        <pipeline-model-definition-plugin.version>2.2081.v3919681ffc1e</pipeline-model-definition-plugin.version>
+        <pipeline-model-definition-plugin.version>2.2084.v1d2999534103</pipeline-model-definition-plugin.version> <!-- TODO need https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/531 for full testing -->
```
Comment (jglick) [inline bom-weekly/pom.xml:15]: https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/531

**Comment by timja** [comment]
Fine to remove 2.289.x at this point imo, we have 5 lines + weekly atm...

**Comment by jglick** [comment]
Another thought I had was to pin the plugin version at `0` in the old line, and update the assertion which requires all managed plugins to be in the sample’s test classpath to ignore any with version `0`.

**Comment by jglick** [comment]
Note that `workflow-cps-global-lib` currently still has a 2.303.x dep, so on the 2.289.x line it is still pinned (#788). This seems to cause some weird behavior in that e.g. https://ci.jenkins.io/job/Tools/job/bom/job/PR-1168/5/testReport/org.jenkinsci.plugins.workflow.libs/FolderLibrariesTest/ runs tests twice on 2.289.x, but appears harmless for now. Could be cleaned up if desired by porting https://github.com/jenkinsci/pipeline-groovy-lib-plugin/pull/5 to the deprecated plugin.

**Review by jtnord — APPROVED** [review]

**Review by dwnusbaum — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Pull request merged**